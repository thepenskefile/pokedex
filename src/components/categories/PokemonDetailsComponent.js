// @flow

import React from 'react';
import { Image, Container, Box, Text, Flex, Tag, Set, Heading, Columns, Column, palette } from 'fannypack';
import Counter from '../../ui/Counter';

type Props = {
  data: Object
};

const PokemonDetailsComponent = ({ data }: Props) => (
  <Container textAlign="left" marginLeft="10px" width="100%" height="70vh" overflowY="scroll" overflowX="hidden">
    <Flex alignItems="flex-end">
      <Heading>{data.name.charAt(0).toUpperCase() + data.name.slice(1)}</Heading>
      <Image src={data.sprites.front_default} />
    </Flex>
    <Set marginBottom="20px" spacing="major-2" display="block">
      <Text use="strong">Height: </Text>
      {data.height}
      <Text use="strong">Weight: </Text>
      {data.weight}
    </Set>
    <Set marginBottom="20px">
      {data.types.map((type, i) => (
        <Tag key={type.type.name} backgroundColor={type.type.name} size="medium">
          {type.type.name}
        </Tag>
      ))}
    </Set>
    <Box marginTop="40px">
      <Heading use="h5">Stats</Heading>
      <Columns marginTop="20px" marginBottom="20px" minBreakpoint="mobile">
        {data.stats.map((stat, i) => (
          <Column key={stat.stat.name} display="inline-block" spreadTablet={4} spreadMobile={6} spreadWidescreen={4}>
            <Counter
              percentage={stat.base_stat}
              label={stat.stat.name}
              color={data.types[0].type.name ? palette(`${data.types[0].type.name}`) : palette('normal')}
            />
          </Column>
        ))}
      </Columns>
    </Box>
    <Box marginTop="40px">
      <Heading use="h5">Sprites</Heading>
      <Columns align="center" textAlign="center">
        {Object.keys(data.sprites).map(
          (keyName, i) =>
            data.sprites[keyName] && (
              <Column
                key={data.name + '_' + keyName}
                display="inline-block"
                minBreakpoint="mobile"
                spread={3}
                spreadMobile={6}
              >
                <Image src={data.sprites[keyName]} />
                <Text display="block">{keyName.replace(/_/g, ' ')}</Text>
              </Column>
            )
        )}
      </Columns>
    </Box>
  </Container>
);

export default PokemonDetailsComponent;
