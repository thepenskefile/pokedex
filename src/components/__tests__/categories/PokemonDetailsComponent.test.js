import React from 'react';
import PokemonDetailsComponent from '../../categories/PokemonDetailsComponent';
import renderer from 'react-test-renderer';

const defaultProps = {
  data: {
    name: 'mockName',
    weight: 'mockWeight',
    height: 'mockHeight',
    sprites: { front_default: 'mockImageUrl' }
  }
};

it('renders correctly', () => {
  const component = renderer.create(<PokemonDetailsComponent {...defaultProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
