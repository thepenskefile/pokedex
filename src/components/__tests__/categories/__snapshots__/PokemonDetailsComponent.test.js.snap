// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="styled-sc-144wh3b-0 iUlbjp Box-sc-6sssmg-0 inpIDs"
  style={
    Object {
      "marginLeft": "10px",
      "textAlign": "left",
      "width": "96%",
    }
  }
  width="96%"
>
  <div
    className="Flex-sp5dv1-0 ddZvmG Box-sc-6sssmg-0 inpIDs"
    style={
      Object {
        "alignItems": "flex-end",
      }
    }
  >
    <h1
      className="heading styled-cya4kb-0 iFEcqF Heading-uhfcou-0 bnhnuY Box-sc-6sssmg-0 inpIDs"
    >
      MockName
    </h1>
    <img
      className="styled-sc-19nzt74-0 ivQgwC Image-idye9d-0 dRjldG Box-sc-6sssmg-0 inpIDs"
      src="mockImageUrl"
    />
  </div>
  <div
    className="Flex-sp5dv1-0 styled-sc-18btutf-0 kkaizR Box-sc-6sssmg-0 inpIDs"
    display="block"
    spacing="major-2"
    style={
      Object {
        "display": "block",
        "marginBottom": "20px",
      }
    }
  >
    <strong
      className="styled-sc-9qokh7-0 skDAJ Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Height: 
    </strong>
    mockHeight
    <strong
      className="styled-sc-9qokh7-0 skDAJ Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Weight: 
    </strong>
    mockWeight
  </div>
  <div
    className="Flex-sp5dv1-0 styled-sc-18btutf-0 kkaizR Box-sc-6sssmg-0 inpIDs"
    spacing="minor-2"
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  >
    <div
      className="Flex-sp5dv1-0 InlineFlex-sc-129poqw-0 styled__Tag-sc-1g01kou-2 gajwsF Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "backgroundColor": "mockTypeName",
        }
      }
    >
      <div
        className="styled__TagContent-sc-1g01kou-1 iuaGjz Box-sc-6sssmg-0 inpIDs"
      >
        mockTypeName
      </div>
    </div>
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
    style={
      Object {
        "marginTop": "40px",
      }
    }
  >
    <h5
      className="heading styled-cya4kb-0 iFEcqF Heading-uhfcou-0 bnhnuY Box-sc-6sssmg-0 inpIDs"
    >
      Stats
    </h5>
    <div
      className="styled-otaduv-0 wkSBA Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "marginBottom": "20px",
          "marginTop": "20px",
        }
      }
    >
      <div
        className="styled__Column-sc-1lmgd0f-0 bXAOxK Box-sc-6sssmg-0 inpIDs"
        display="inline-block"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <div
          className="Box-sc-6sssmg-0 inpIDs"
        >
          <svg
            className="sc-bdVaJa fJZkWb"
            viewBox="0 0 36 36"
          >
            <path
              className="sc-bwzfXH fUxNdF"
              d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
            />
            <path
              className="sc-bxivhb boxNwV"
              color={[Function]}
              d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
            />
            <text
              className="sc-htpNat jJPiwm"
              x="18"
              y="20.35"
            >
              mockBaseStat
            </text>
          </svg>
          <div
            className="Box-sc-6sssmg-0 inpIDs"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            mockStatName
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
    style={
      Object {
        "marginTop": "40px",
      }
    }
  >
    <h5
      className="heading styled-cya4kb-0 iFEcqF Heading-uhfcou-0 bnhnuY Box-sc-6sssmg-0 inpIDs"
    >
      Sprites
    </h5>
    <div
      className="styled-otaduv-0 wkSBA Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="styled__Column-sc-1lmgd0f-0 jUBHv Box-sc-6sssmg-0 inpIDs"
        display="inline-block"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          className="styled-sc-19nzt74-0 ivQgwC Image-idye9d-0 dRjldG Box-sc-6sssmg-0 inpIDs"
          src="mockImageUrl"
        />
        <span
          className="styled-sc-9qokh7-0 skDAJ Box-sc-6sssmg-0 inpIDs"
          display="block"
          style={
            Object {
              "color": undefined,
              "display": "block",
            }
          }
        >
          front default
        </span>
      </div>
    </div>
  </div>
</div>
`;
